src/main/java/org/logstash/plugins/inputs/http/util/SslSimpleBuilder.java

```diff
@@ -0,0 +1,193 @@
+        FORCE_PEER,
+    }
+    private final static Logger logger = LogManager.getLogger(SslSimpleBuilder.class);
+
+
+    private File sslKeyFile;
+    private File sslCertificateFile;
+    private SslClientVerifyMode verifyMode = SslClientVerifyMode.FORCE_PEER;
+
+    private long handshakeTimeoutMilliseconds = 10000;
+
+    /*
+    Modern Ciphers Compatibility List from
+    https://wiki.mozilla.org/Security/Server_Side_TLS
+    This list require the OpenSSl engine for netty.
+    */
+    public final static String[] DEFAULT_CIPHERS = new String[] {
+            "TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384",
+            "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384",
+            "TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256",
+            "TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
+            "TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384",
+            "TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384",
+            "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256",
+            "TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256"
+    };
+
+    private String[] ciphers = DEFAULT_CIPHERS;
+    private String[] protocols = new String[] { "TLSv1.2" };
+    private String[] certificateAuthorities;
+    private String passPhrase;
+
+    public SslSimpleBuilder(String sslCertificateFilePath, String sslKeyFilePath, String pass) throws FileNotFoundException {
+        sslCertificateFile = new File(sslCertificateFilePath);
+        sslKeyFile = new File(sslKeyFilePath);
+        passPhrase = pass;
+        ciphers = DEFAULT_CIPHERS;
+    }
+
+    public SslSimpleBuilder setProtocols(String[] protocols) {
+        this.protocols = protocols;
+        return this;
+    }
+
+    public SslSimpleBuilder setCipherSuites(String[] ciphersSuite) throws IllegalArgumentException {
+        for(String cipher : ciphersSuite) {
+            if(!OpenSsl.isCipherSuiteAvailable(cipher)) {
+                throw new IllegalArgumentException("Cipher `" + cipher + "` is not available");
+            } else {
+                logger.debug("Cipher is supported: " + cipher);
+            }
+        }
+
+        ciphers = ciphersSuite;
+        return this;
+    }
+
+    public SslSimpleBuilder setCertificateAuthorities(String[] cert) {
+        certificateAuthorities = cert;
+        return this;
+    }
+
+    public SslSimpleBuilder setHandshakeTimeoutMilliseconds(int timeout) {
+        handshakeTimeoutMilliseconds = timeout;
+        return this;
+    }
+
+    public SslSimpleBuilder setVerifyMode(SslClientVerifyMode mode) {
+        verifyMode = mode;
+        return this;
+    }
+
+    public File getSslKeyFile() {
+        return sslKeyFile;
+    }
+
+    public File getSslCertificateFile() {
+        return sslCertificateFile;
+    }
+
+    public SslHandler build(ByteBufAllocator bufferAllocator) throws IOException, NoSuchAlgorithmException, CertificateException {
+        SslContextBuilder builder = SslContextBuilder.forServer(sslCertificateFile, sslKeyFile, passPhrase);
+
+        if(logger.isDebugEnabled())
```

Comment:
were both logging statements intended to be inside the if(logger.isDebugEnabled() ?

parameterized messages are preferable to concats for logging (and often avoids the need for the if(logger.is<>Enabled) check.  For example `logger.debug("Ciphers {}", Arrays.toString(ciphers))`

Also, there is a mix of if`\n`X and if {`\n`X`\n` } , I don't want to start a war, but we should be consistent.